CRM/Admin/Form/Setting.php

```diff
@@ -133,7 +133,12 @@ public function buildQuickForm() {
           $this->add('date', $setting, ts($props['title']), CRM_Core_SelectValues::date(NULL, 'M d'));
         }
         else {
-          $this->$add($setting, ts($props['title']));
+          if (CRM_Core_BAO_Setting::isEnvironmentSet($setting)) {
```

Comment:
The other settings do not check if they have been explicitly set in civicrm.settings.php. In our implementation, we check first, and then disable the widget. If we have to adopt a more generalized approach, then we should disable the widgets for all the settings specified in civicrm.settings.php for this form.